src/boot/efi/boot.c

```diff
@@ -610,28 +610,16 @@ static BOOLEAN menu_run(
                                         uefi_call_wrapper(ST->ConOut->SetAttribute, 2, ST->ConOut,
                                                           EFI_LIGHTGRAY|EFI_BACKGROUND_BLACK);
                                 uefi_call_wrapper(ST->ConOut->OutputString, 2, ST->ConOut, lines[i]);
-                                if ((INTN)i == config->idx_default_efivar) {
-                                        uefi_call_wrapper(ST->ConOut->SetCursorPosition, 3, ST->ConOut, x_start-3, y_start + i - idx_first);
-                                        uefi_call_wrapper(ST->ConOut->OutputString, 2, ST->ConOut, (CHAR16*) L"=>");
-                                }
                         }
                         refresh = FALSE;
                 } else if (highlight) {
                         uefi_call_wrapper(ST->ConOut->SetCursorPosition, 3, ST->ConOut, 0, y_start + idx_highlight_prev - idx_first);
                         uefi_call_wrapper(ST->ConOut->SetAttribute, 2, ST->ConOut, EFI_LIGHTGRAY|EFI_BACKGROUND_BLACK);
                         uefi_call_wrapper(ST->ConOut->OutputString, 2, ST->ConOut, lines[idx_highlight_prev]);
-                        if ((INTN)idx_highlight_prev == config->idx_default_efivar) {
-                                uefi_call_wrapper(ST->ConOut->SetCursorPosition, 3, ST->ConOut, x_start-3, y_start + idx_highlight_prev - idx_first);
-                                uefi_call_wrapper(ST->ConOut->OutputString, 2, ST->ConOut, (CHAR16*) L"=>");
-                        }
 
                         uefi_call_wrapper(ST->ConOut->SetCursorPosition, 3, ST->ConOut, 0, y_start + idx_highlight - idx_first);
                         uefi_call_wrapper(ST->ConOut->SetAttribute, 2, ST->ConOut, EFI_BLACK|EFI_BACKGROUND_LIGHTGRAY);
                         uefi_call_wrapper(ST->ConOut->OutputString, 2, ST->ConOut, lines[idx_highlight]);
-                        if ((INTN)idx_highlight == config->idx_default_efivar) {
-                                uefi_call_wrapper(ST->ConOut->SetCursorPosition, 3, ST->ConOut, x_start-3, y_start + idx_highlight - idx_first);
-                                uefi_call_wrapper(ST->ConOut->OutputString, 2, ST->ConOut, (CHAR16*) L"=>");
-                        }
```

Comment:
i don't follow? i think we need to communicate somehow what was set before as manual "default": dropping this sounds weird.

i mean we can certainly show this differently, i.e. say "default" or so, to make this clearerer. But hiding this altogether?

(i#d also be fine with showing this in prose always at the bottom or so)

Reply:
This highlights whatever the efivar default *currently* is. This does *never* highlight the value from loader.conf. So if you boot without the efivar you never see this. Meanwhile, if you have it set on boot, the default will be selected anyway, so we don't really need to highlight it like this at all (imho).

I think you might be confusing this with changing it at runtime with d/D. If you press d/D at runtime, you have this entry selected anyway, making it unnecessary there too. And the status line already says that the default was changed.

Reply:
> This highlights whatever the efivar default _currently_ is. This does _never_ highlight the value from loader.conf. So if you boot without the efivar you never see this.

Yes, exactly. That's a good thing. We only show this if the user made a manual change via the UI.

The changes the user can make in the UI should be considered *user* changes, and the ones via loader.conf are more in admin/vendor territory. I think it's worth showing that the *user* overrode what the admin/vendor set here, and that's what => indicates.

Reply:
I'm still not convinced. The information is there in print_status as well as by the fact that it it selected automatically at boot. The highlighting of the efivar default is simply superfluous and looks terrible to me.

The same argument could be had about the timeout. That one can be changed with the UI too, yet does not get special treatment. And the upcoming console mode efivar.

Reply:
if in the user UI you can change something (and you can by hitting d) we need to communicate its effect somehow. Maybe the arrow is not the clearest way, but not showing it at all is worse. Otherwise people will keep hitting d and never know whether it did anything, and what was it set to last.

Reply:
> we need to communicate its effect somehow

But it does that already. It shows a status line saying the default entry has been set/cleared.

Reply:
so, but when i boot, how do i figure out what i previously set? (and no, don't tell me to go to this wall of text status dump)

Reply:
Well, next time you boot, the thing you previously set will be selected by default. On the other hand, if you cleared the default the last time, you couldn't get a highlight anyway. I really don't see the value of this highlight.

Reply:
if i want to see the boot menu i keep pressing up/down keys during boot. that will pop up the menu, but also likely select a different menu item already, and i have no idea what was actually set by default, i.e. what would have ended up being booted if i hadn't hit up/down keys

Reply:
So how about only showing this if menu is disabled/timeout == 0? Btw, using the spacebar is a much better choice for this :P

Reply:
note that we don't show this normally anyway – until you actually hit "d" to make something the default. Hence, it's not really clutter that annoys people, it's something we only show if people basically explicitly asked for it.

again, i am fine to show this some other way, but i still think we should show it, if the user ever set something.